src/github.com/matrix-org/dendrite/roomserver/storage/storage.go

```diff
@@ -431,6 +446,20 @@ func (d *Database) LatestEventIDs(
 	return references, currentStateSnapshotNID, depth, nil
 }
 
+func (d *Database) GetRoomsForUserMembership(
+	ctx context.Context,
+	userNID types.EventStateKeyNID,
+	membership string,
+) (roomNIDs []types.RoomNID, err error) {
```

Comment:
Is there a spec which defines this as fallback? I would expect an error when I trigger a query with an unknown membership type.

Reply:
I don't think so, however Dendrite doesn't have all the membership states split out yet. A `leave || ban` check would probably be best.